src/overmap.cpp

```diff
@@ -3778,6 +3808,17 @@ std::shared_ptr<npc> overmap::find_npc( const int id ) const
     return nullptr;
 }
 
+bool overmap::is_omt_generated( const tripoint &loc ) const
+{
+    if( !inbounds( loc ) ) {
+        return false;
+    }
+
+    const bool is_generated = MAPBUFFER.lookup_submap( omt_to_sm_copy( loc ) ) != nullptr;
```

Comment:
This does not match. `inbounds` above takes local coordinates, but `MAPBUFFER::lookup_submap` requires a global variable. You need to add `overmap::loc` (converted to the correct coordinate system).